diff --git a/Include/cpython/pystate.h b/Include/cpython/pystate.h
index c37123c4f6..aa518281c8 100644
--- a/Include/cpython/pystate.h
+++ b/Include/cpython/pystate.h
@@ -56,7 +56,7 @@ typedef struct _err_stackitem {
      * This ensures that the exception state is not impacted by "yields"
      * from an except handler.
      */
-    PyObject *exc_value;
+    PyObject *exc_type, *exc_value, *exc_traceback;
 
     struct _err_stackitem *previous_item;
 
diff --git a/i386/Modules/Setup.local b/i386/Modules/Setup.local
new file mode 100644
index 0000000000..468bd1f1c1
--- /dev/null
+++ b/i386/Modules/Setup.local
@@ -0,0 +1,34 @@
+*static* 
+    array arraymodule.c 
+    binascii binascii.c 
+    cmath cmathmodule.c 
+    math mathmodule.c 
+    pyexpat pyexpat.c 
+    select selectmodule.c 
+    unicodedata unicodedata.c 
+    zlib  zlibmodule.c 
+    _asyncio _asynciomodule.c 
+    _blake2 _blake2/blake2module.c _blake2/blake2b_impl.c _blake2/blake2s_impl.c 
+    _ctypes _ctypes/_ctypes.c _ctypes/callbacks.c _ctypes/callproc.c _ctypes/stgdict.c _ctypes/cfield.c -ldl -lffi 
+    _contextvars _contextvarsmodule.c 
+    _datetime _datetimemodule.c 
+    _decimal _decimal/_decimal.c 
+    _elementtree _elementtree.c 
+    _hashlib _hashopenssl.c -lcrypto 
+    _heapq _heapqmodule.c 
+    _json _json.c 
+    _md5 md5module.c 
+    _multibytecodec cjkcodecs/multibytecodec.c 
+    _multiprocessing -I/opt/cpython/Modules/_multiprocessing _multiprocessing/multiprocessing.c _multiprocessing/semaphore.c 
+    _opcode _opcode.c 
+    _pickle _pickle.c 
+    _queue _queuemodule.c 
+    _random _randommodule.c 
+    _sha1 sha1module.c 
+    _sha256 sha256module.c 
+    _sha512 sha512module.c 
+    _sha3 _sha3/sha3module.c 
+    _socket socketmodule.c 
+    _ssl _ssl.c -lssl 
+    _struct _struct.c 
+    
